plugins/catalog-backend-module-ldap/src/ldap/client.ts

```diff
@@ -58,37 +57,29 @@ export class LdapClient {
       });
     }
 
-    const client = ldap.createClient({
+    const client = new Client({
       url: target,
       tlsOptions: {
         secureContext,
         rejectUnauthorized: tls?.rejectUnauthorized,
       },
```

Comment:
After some debugging, I realized that LDAP wasn't working on my end because of [this check](https://github.com/ldapts/ldapts/blob/f117fb0aec0a3ce62f2e10ad7690b7fa41bbae2c/src/Client.ts#L182).


That expression evaluates to true whenever `tlsOptions` is an object, even if all its properties are undefined:
```
{
  secureContext: undefined,
  rejectUnauthorized: undefined,
}
```
As a result, the client behaves as if TLS options were provided and the connection is expected to be secure. We should instead pass undefined when no TLS configs are present.

Reply:
> @ganievs Thanks so much for addressing this and creating the issue! I tested again, but the same issue persists :(
> 
> `Object.keys(tlsOptions).length > 0` evaluates to true since it's checking for the length of the keys in the object and `tlsOptions` is created with keys even when all values are undefined. I think in this case we can do something like this that checks for any defined values:
> 
> ```
> ...(Object.values(tlsOptions).some(v => v !== undefined) ? { tlsOptions } : undefined)
> ```
Fixed with your suggestion - `Object.values(tlsOptions).some(v => v !== undefined)`. Сould you confirm if this solved the problem? Thanks!

Reply:
It works perfectly now, thanks for addressing this!